src/permission/fs_permission.h

```diff
@@ -32,12 +32,12 @@ class FSPermission final : public PermissionBase {
 
       Node() : wildcard_child(nullptr), is_leaf(false) {}
 
-      Node* CreateChild(const std::string& prefix) {
-        if (prefix.empty() && !is_leaf) {
+      Node* CreateChild(const std::string& path_prefix) {
+        if (path_prefix.empty() && !is_leaf) {
           is_leaf = true;
           return this;
         }
-        char label = prefix[0];
+        char label = path_prefix[0];
```

Comment:
Should we add an `CHECK(!path_prefix.empty())` to here? If path is empty and if it is `is_leaf = true`, this would throw an exception

Reply:
sure, happy to add that!